solidity/contracts/PetalLock.sol

```diff
@@ -0,0 +1,136 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.26;
+
+// ENS contracts
+address constant ENS_PUBLIC_RESOLVER = 0x231b0Ee14048e9dCcD1d247744d114a4EB5E8E63;
+address constant ENS_TOKEN_WRAPPER = 0xD4416b13d2b3a9aBae7AcD5D6C2BbDBE25686401;
+address constant ENS_REGISTRY_WITH_FALLBACK = 0x00000000000C2E074eC69A0dFb2997BA6C7d2e1e;
+
+// ENS Fuses
+uint16 constant CANNOT_UNWRAP = 1;
+uint16 constant CANNOT_BURN_FUSES = 2;
+uint16 constant CANNOT_TRANSFER = 4;
+uint16 constant CANNOT_SET_RESOLVER = 8;
+uint16 constant CANNOT_SET_TTL = 16;
+uint16 constant CANNOT_CREATE_SUBDOMAIN = 32;
+uint16 constant CANNOT_APPROVE = 64;
+uint32 constant PARENT_CANNOT_CONTROL = 1 << 16;
+uint32 constant IS_DOT_ETH = 1 << 17;
+uint32 constant CAN_EXTEND_EXPIRY = 1 << 18;
+uint16 constant CAN_DO_EVERYTHING = 0;
+
+// interfaces
+interface IEnsRegistryWithFallBack {
+	function recordExists(bytes32 node) external view returns (bool);
+}
+
+interface IEnsTokenWrapper {
+	function safeTransferFrom(address from, address to, uint256 tokenId, uint256 amount, bytes memory _data) external;
+	function safeBatchTransferFrom(address from, address to, uint256[] memory tokenId, uint256[] memory amount, bytes memory _data) external;
+	function setSubnodeRecord(bytes32 parentNode, string memory label, address owner, address resolver, uint64 ttl, uint32 fuses, uint64 expiry) external returns(bytes32 node);
+	function getData(uint256 id) external view returns (address, uint32, uint64);
+	function setChildFuses(bytes32 parentNode, bytes32 labelhash, uint32 fuses, uint64 expiry) external;
+	function isWrapped(bytes32 node) external view returns (bool);
+	function setFuses(bytes32 node, uint16 fuses) external returns (uint32);
+}
+
+interface IEnsPublicResolver {
+	function setContenthash(bytes32 node, bytes calldata hash) external;
+}
+
+IEnsRegistryWithFallBack constant ensRegistry = IEnsRegistryWithFallBack(ENS_REGISTRY_WITH_FALLBACK);
+IEnsTokenWrapper constant ensTokenWrapper = IEnsTokenWrapper(ENS_TOKEN_WRAPPER);
+IEnsPublicResolver constant ensPublicResolver = IEnsPublicResolver(ENS_PUBLIC_RESOLVER);
+
+// fuse combinations
+uint32 constant PARENT_FUSES_TO_BURN = CANNOT_UNWRAP + PARENT_CANNOT_CONTROL;
+uint32 constant FINAL_CHILD_FUSES_TO_BURN = CANNOT_UNWRAP + CANNOT_BURN_FUSES + CANNOT_SET_RESOLVER + CANNOT_SET_TTL + CANNOT_CREATE_SUBDOMAIN + PARENT_CANNOT_CONTROL + CANNOT_APPROVE + CAN_EXTEND_EXPIRY;
+address constant burnAddress = 0xdeaDDeADDEaDdeaDdEAddEADDEAdDeadDEADDEaD;
```

Comment:
```suggestion
address constant BURN_ADDRESS = 0xdeaDDeADDEaDdeaDdEAddEADDEAdDeadDEADDEaD;
```
To match constant casing elsewhere.  Will need to find and fix sites where it is referenced as well.